### PR TITLE
Add stop-task event in order to inform Marathon-lb which containers are about to be killed

### PR DESCRIPTION
**Pull request by mushixun**
When using marathon-lb for load balancing, when marathon is shrinking, some requests fail. After analysis, it is because the container has not been removed from the haproxy routing table when the container is killed. It is not caused by haproxy reload, so I hope marathon can tell which container to shrink in advance when shrinking. So that I can first remove the container information from haproxy's routing table, no longer accept new requests, and then kill the container, and the premise of this function is that marathon needs to issue the first stop when shrinking. send stop event information.Tell marathon-lb that which containers will be stop.

JIRA issues: MARATHON-8339

### CODE REVIEW

File: src/main/scala/mesosphere/marathon/api/v2/json/Formats.scala [hunk 1]
```diff
@@ -546,4 +557,4 @@ trait PluginFormats {
   ) (d => (d.id, d.plugin, d.implementation, d.tags, d.info))
 
   implicit lazy val pluginDefinitionsFormat: Writes[PluginDefinitions] = Json.writes[PluginDefinitions]
-}
+}
```
Comment (ichernetsky) [inline src/main/scala/mesosphere/marathon/api/v2/json/Formats.scala]: It seems your editor is configured to drop trailing empty lines. Please make sure there are no such changes throughout this PR.

Comment (mushixun) [reply src/main/scala/mesosphere/marathon/api/v2/json/Formats.scala]: yes，it no change，maybe my editor wrong。

Comment (mushixun) [reply src/main/scala/mesosphere/marathon/api/v2/json/Formats.scala]: need i to re commit？

Comment (ichernetsky) [reply src/main/scala/mesosphere/marathon/api/v2/json/Formats.scala]: Not re-commit, but push a new commit on top of the existing one in this branch.

**Comment by mushixun** [comment]
what error about check?

**Comment by timcharper** [comment]
Right now, with you change, I see this:

```json
event: deployment_info
data: {"plan":{"id":"fe417996-26a6-42fc-8d0d-bf30fc17668c","steps":[{"actions":[{"action":"StopApplication","app":"/sleeper"}]}],"version":"2018-09-06T16:36:17.621Z"},"currentStep":{"actions":[{"action":"StopApplication","app":"/sleeper"}]},"eventType":"deployment_info","timestamp":"2018-09-06T16:36:17.634Z"}

event: group_change_success
data: {"groupId":"/","version":"2018-09-06T16:36:17.621Z","eventType":"group_change_success","timestamp":"2018-09-06T16:36:17.644Z"}

event: stop_task_event
data: {"taskId":["sleeper.ee192f88-b1f2-11e8-b355-8efa78a6090e"],"instanceId":"sleeper.marathon-ee192f88-b1f2-11e8-b355-8efa78a6090e","eventType":"stop_task_event","timestamp":"2018-09-06T16:36:17.687Z"}

event: status_update_event
data: {"slaveId":"ab6ef384-5970-4875-b361-48daf05d5276-S0","taskId":"sleeper.ee192f88-b1f2-11e8-b355-8efa78a6090e","taskStatus":"TASK_KILLED","message":"Command terminated with signal Terminated: 15","appId":"/sleeper","host":"bastion.lan","ipAddresses":[{"ipAddress":"10.0.5.8","protocol":"IPv4"}],"ports":[31102],"version":"2018-09-06T16:35:56.564Z","eventType":"status_update_event","timestamp":"2018-09-06T16:36:17.827Z"}

event: instance_changed_event
data: {"instanceId":"sleeper.marathon-ee192f88-b1f2-11e8-b355-8efa78a6090e","condition":"Killed","runSpecId":"/sleeper","agentId":"ab6ef384-5970-4875-b361-48daf05d5276-S0","host":"bastion.lan","runSpecVersion":"2018-09-06T16:35:56.564Z","timestamp":"2018-09-06T16:36:17.828Z","eventType":"instance_changed_event"}
```

I propose we instead publish this:

```diff
--- before	2018-09-06 10:40:17.000000000 -0600
+++ after	2018-09-06 10:41:09.000000000 -0600
@@ -4,8 +4,8 @@
 event: group_change_success
 data: {"groupId":"/","version":"2018-09-06T16:36:17.621Z","eventType":"group_change_success","timestamp":"2018-09-06T16:36:17.644Z"}
 
-event: stop_task_event
-data: {"taskId":["sleeper.ee192f88-b1f2-11e8-b355-8efa78a6090e"],"instanceId":"sleeper.marathon-ee192f88-b1f2-11e8-b355-8efa78a6090e","eventType":"stop_task_event","timestamp":"2018-09-06T16:36:17.687Z"}
+event: instance_changed_event
+data: {"instanceId":"sleeper.marathon-ee192f88-b1f2-11e8-b355-8efa78a6090e","condition":"Killing","runSpecId":"/sleeper","agentId":"ab6ef384-5970-4875-b361-48daf05d5276-S0","host":"bastion.lan","runSpecVersion":"2018-09-06T16:35:56.564Z","timestamp":"timestamp":"2018-09-06T16:36:17.827Z","eventType":"instance_changed_event"}
 
 event: status_update_event
 data: {"slaveId":"ab6ef384-5970-4875-b361-48daf05d5276-S0","taskId":"sleeper.ee192f88-b1f2-11e8-b355-8efa78a6090e","taskStatus":"TASK_KILLED","message":"Command terminated with signal Terminated: 15","appId":"/sleeper","host":"bastion.lan","ipAddresses":[{"ipAddress":"10.0.5.8","protocol":"IPv4"}],"ports":[31102],"version":"2018-09-06T16:35:56.564Z","eventType":"status_update_event","timestamp":"2018-09-06T16:36:17.827Z"}
```

**Comment by mushixun** [comment]
OK,killing condition is ready?@timcharper

**Pull request closed without merging**